Xamarin.Forms.Platform.iOS/CollectionView/ObservableGroupedSource.cs

```diff
@@ -117,66 +90,86 @@ void ClearGroupTracking()
 			}
 		}
 
-		void ResetGroupTracking()
+		void ResetGroups()
 		{
-			ClearGroupTracking();
+			ClearGroups();
 
-			for (int n = 0; n < _groupSource.Count; n++)
+			var index = 0;
+			foreach (var group in _originalItemsSource)
 			{
-				if (_groupSource[n] is INotifyCollectionChanged && _groupSource[n] is IEnumerable list)
+				if (group is IEnumerable list)
+				{
+					if (group is INotifyCollectionChanged)
+					{
+						_groups.Add(new ObservableItemsSource(list, _collectionViewController, index));
+					}
+					else
+					{
+						_groups.Add(new ListSource(list, index));
+					}
+				}
+				else
 				{
-					_groups.Add(new ObservableItemsSource(list, _collectionViewController, n));
+					_groups.Add(new ListSource(index));
```

Comment:
This is a really unlikely edge case, but I still want to make sure that we add an item to `_groups` in any case, even when it is not a list (for whatever reason), I add an empty list here.